### PR TITLE
fix(typescript) better highlighting of optionals

### LINKED ISSUE #3613: (typescript) optional property not highlighted correctly
**Describe the issue**
 optional property (`prop?: number`) not highlighted correctly.

**Which language seems to have the issue?**
TypeScript

**Are you using `highlight` or `highlightAuto`?**
highlight? Using highlightjs through revealjs VSCode extension.

**Sample Code to Reproduce**

```ts
type Options = {
  srcdir: string,
  outdir: string,
  minify?: boolean
}
```

Result (using Github Dark):
image

**Expected behavior**

As seen above, rendered by Github

### PR DESCRIPTION
**Pull request by Dxuian**
Fixed the optional property in ts not being highlighted 
Resolves #3613
before:
image

after: 
image

### Checklist
- [x] Added markup tests, or they don't apply here because...
- [x] Updated the changelog at `CHANGES.md`

### CODE REVIEW

File: src/languages/lib/ecmascript.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export const IDENT_RE = '[A-Za-z$_][0-9A-Za-z$_]*';
+export const IDENT_RE = '[A-Za-z$_][0-9A-Za-z$_]*\\??';
```
Comment (joshgoebel) [inline src/languages/lib/ecmascript.js:1]: Is `?` truly a valid ECMAScript identifier, or just in TS?

File: src/languages/typescript.js [hunk 2]
```diff
@@ -69,19 +69,22 @@ export default function(hljs) {
     "satisfies"
   ];
 
-  /*
-    namespace is a TS keyword but it's fine to use it as a variable name too.
-    const message = 'foo';
-    const namespace = 'bar';
-  */
-
+  const OPTIONAL_PROPERTY_RE = IDENT_RE + '\\s*\\?';
```
Comment (joshgoebel) [inline src/languages/typescript.js:72]: Please provide an example of what this is trying to do.

File: src/languages/lib/ecmascript.js (hunk 1 above)
Comment (Dxuian) [reply src/languages/lib/ecmascript.js:1]: Youre right , ive managed to scope the new fix to just ts in the new commit

File: src/languages/typescript.js (hunk 2 above)
Comment (Dxuian) [reply src/languages/typescript.js]: in my new commit ive added some test so i found that it was failing on some new tcs so ill explain the new one

**Comment by joshgoebel** [comment]
I rewrote this to use more of the existing `attr` rule.  All tests still seem to pass.  See any issues?

**Comment by Dxuian** [comment]
> I rewrote this to use more of the existing `attr` rule. All tests still seem to pass. See any issues?

looks good to me can we close and merge this issue ?

**Comment by Dxuian** [comment]
> I rewrote this to use more of the existing `attr` rule. All tests still seem to pass. See any issues?

its highlighting wrong 
image
before:
image

i dont see what seems to be the issue in the previous commit i made  can you ellaborate a little more

**Comment by joshgoebel** [comment]
> its highlighting wrong

The tests [you provided] (including the one you show above) are all still green... _given that the tests are all green, my version has the exact same behavior as yours_.  So... not sure what you are seeing.

```
# ./tools/build.js -t node && npm run test-markup
...
 531 passing (1s)
```

> i dont see what seems to be the issue in the previous commit i made 

It's just a bit too complex... we usually try for the simplest solutions.

**Comment by Dxuian** [comment]
> > its highlighting wrong
> 
> The tests [you provided] (including the one you show above) are all still green... _given that the tests are all green, my version has the exact same behavior as yours_. So... not sure what you are seeing.
> 
> ```
> # ./tools/build.js -t node && npm run test-markup
> ...
>  531 passing (1s)
> ```
> 
> > i dont see what seems to be the issue in the previous commit i made
> 
> It's just a bit too complex... we usually try for the simplest solutions.

ah my apologies i changed to a different style and thought comment were being highlighted wrong 
this seems to work fine can you merge and close this issue ?

**Review by joshgoebel — APPROVED** [review]

**Review by Theghost256 — APPROVED** [review]

**Pull request merged**